### PR TITLE
feat(schemas): add roles and permissions

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add RBAC related tables:

1. Add table `scopes`, with a foreign key pointing to `resources`
2. Add `id` to `roles`, because we have a display name now, and judge access by `role_name` is deperecated.
3. Add table `roles_scopes` to implement many-to-many relations.

After consideration and research, instead of `jsonb`, use the traditional foreign key constraints and a "relation" table, that would be more flexable and efficency: https://dba.stackexchange.com/questions/82009/postgres-json-field-for-many-to-many-relationships

## Testing

/

### CODE REVIEW

File: packages/schemas/tables/roles.sql [hunk 1]
```diff
@@ -1,5 +1,6 @@
 create table roles (
-    name varchar(128) not null,
-    description varchar(128) not null,
-    primary key (name)
+  id varchar(21) not null,
+  name varchar(128) not null,
+  description varchar(128) not null,
+  primary key (id)
```
Comment (gao-sun) [inline packages/schemas/tables/roles.sql:5]: name should be unique as well?

File: packages/schemas/tables/scopes.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+create table scopes (
+  id varchar(21) not null,
+  resource_id varchar(21) not null,
+  name varchar(256) not null,
+  description text,
+  created_at timestamptz not null default(now()),
+  primary key (id),
+  constraint fk_scopes_resource_id
```
Comment (gao-sun) [inline packages/schemas/tables/scopes.sql:8]: will it be better to follow the same format as indices?
```suggestion
  constraint fk__scopes__resource_id
```

File: packages/schemas/tables/roles.sql (hunk 1 above)
Comment (wangsijie) [reply packages/schemas/tables/roles.sql:5]: Unique index added.

File: packages/schemas/alterations/next-1669091623-roles-and-scopes.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        created_at timestamptz not null default(now()),
+        primary key (id)
+      );
+      -- update table roles, add id and replace pkey
+      alter table roles add column id varchar(21);
+      update roles set id = name;
+      alter table roles alter column id set not null;
+      alter index roles_pkey rename to roles_pkey_1;
+      create unique index roles_pkey on roles using btree(id)
+      drop index roles_pkey_1;
```
Comment (gao-sun) [inline packages/schemas/alterations/next-1669091623-roles-and-scopes.ts:23]: no need for this?

File: packages/schemas/tables/scopesroles.sql [hunk 4]
```diff
@@ -0,0 +1,5 @@
+create table roles_scopes (
+  role_id varchar(21) references roles (id) on update cascade on delete cascade,
+  scope_id varchar(21) references scopes (id) on update cascade on delete cascade,
+  constraint roles_permissison_pkey primary key (role_id, scope_id)
```
Comment (gao-sun) [inline packages/schemas/tables/scopesroles.sql:4]:
```suggestion
  primary key (role_id, scope_id)
```

File: packages/schemas/alterations/next-1669091623-roles-and-scopes.ts (hunk 3 above)
Comment (wangsijie) [reply packages/schemas/alterations/next-1669091623-roles-and-scopes.ts]: Need to change prev "primary key index" to a normal unique index, updated with more straitforword queries. PTAL

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**